Common/Util/StreamReaderExtensions.cs

```diff
@@ -76,6 +76,46 @@ public static decimal GetDecimal(this StreamReader stream, char delimiter = Defa
             return new decimal(lo, mid, 0, isNegative, (byte)(hasDecimals ? decimalPlaces : 0));
         }
 
+        /// <summary>
+        /// Reads the next string value from the CSV file
+        /// </summary>
+        /// <param name="stream">The data stream</param>
+        /// <param name="delimiter">The data delimiter character to use, default is ','</param>
+        /// <returns>The string value read from CSV</returns>
+        public static string GetNextCsv(this StreamReader stream, char delimiter = DefaultDelimiter)
+        {
```

Comment:
It is not clear to me the reason for this extension method, it is only used here:

https://github.com/QuantConnect/Lean/blob/7fd9862eb1ab69f285e0d50e6acc623ad5df30eb/Common/Data/Custom/TradingEconomics/TradingEconomicsCalendar.cs#L249

Why don't simply use `streamReader.ReadLine().ToCsvData()` in `TradingEconomicsCalendar.Reader()`?

Reply:
This implementation has added benefits in being able to parse values enclosed in quotes. I do plan on using that implementation in the future for other data sources 😄

Reply:
`ToCsvData` is able to parse values enclosed in quotes. 

Unless better criteria, I would recommend removing this method and its usage in `TradingEconomicsCalendar.Reader()`. 

Any thoughts @Martin-Molinero?

Reply:
There is one benefit to the new method, and that is that we completely let go of the integer indexing style of reading string values from the CSV.

So for example, if we have this data and it changes to the new format:
Old:
```
20190101 00:00:00,20190101 00:00:00,hello,world!
```

New:
```
hello,20190101 00:00:00,world!,20190101 00:00:00
```

The old way of putting it into code would be like this (using ToCsvData(...)):

```
fn Reader(...)
{
    var data = reader.ReadLine().ToCsvData();

    instance.Time = Parse.Exact(data[0], "yyyyMMdd HH:mm:ss");
    instance.EndTime = Parse.Exact(data[1], "yyyyMMdd HH:mm:ss");
    instance.Text = data[2];
    instance.Last = data[3];
}
```

But would have to be done like this after the changes:

```
fn Reader(...)
{
    var data = reader.ReadLine().ToCsvData();

    instance.Text = data[0];
    instance.Time = Parse.Exact(data[1], "yyyyMMdd HH:mm:ss");
    instance.Last = data[2];
    instance.EndTime = Parse.Exact(data[3], "yyyyMMdd HH:mm:ss");
}
```

Which can be very error prone when updating the number of the CSV index!

However, if we use the new `GetNextCsv()` method, we can rewrite the above to the following:
Old:
```
fn Reader(...)
{
    instance.Time = reader.GetDateTime("yyyyMMdd HH:mm:ss");
    instance.EndTime = reader.GetDateTime("yyyyMMdd HH:mm:ss");
    instance.Text = reader.GetNextCsv();
    instance.Last = reader.GetNextCsv();
}
```

New:
```
fn Reader(...)
{
    instance.Text = reader.GetNextCsv();
    instance.Time = reader.GetDateTime("yyyyMMdd HH:mm:ss");
    instance.Last = reader.GetNextCsv();
    instance.EndTime = reader.GetDateTime("yyyyMMdd HH:mm:ss");
}
```
All it required was just switching the lines around to the order we encountered them.

Given how much I've messed up CSV indexes in the past, I'd much rather prefer using `GetNextCsv()` 😄

Reply:
As always, you have a sound reason why.

My main concerns are readability and maintainability of code that - in principle - makes a task already implemented.

Reply:
The main value of the `StreamReader` is performance. `GetNextCsv()` will parse the line and generate the final strings in the same operation, while `ReadLine()` + `ToCsvData()` will read the line, generate a new string, later parse that string again and generate the final strings. We know this makes the `StreamReader` ~50% faster. More at https://github.com/QuantConnect/Lean/pull/3857 . **But its more complex**, I think I agree with @Jay-Jay-D, if there isn't a measurable performance gain, the added complexity is not worth the trouble. The gain will be measurable from the Lean user perspective with big data sets.
@gsalaz98 regarding the indexing, maybe it could use and `int index = 0`, which gets incremented on each access `[index++]`